### PR TITLE
fix: match Chrome's font fallback behavior

### LINKED ISSUE #15481: Default fonts should match Chrome
Chrome has default font fallbacks set up on a per-script, per-OS basis. See for example https://chromium.googlesource.com/chromium/src/+/master/chrome/app/resources/locale_settings_mac.grd#160, which is the default font for characters in the Japanese-language script.

Electron does not set up these font fallbacks, and as a result, font rendering in those scripts does not match that of Chrome. Further, it's impossible to set it up on the app side to use the appropriate font fallbacks, because of #15480.

### PR DESCRIPTION
**Pull request by nornagon**
This PR sets up blink's default fonts (`serif`, `sans-serif`, etc.) to match Chrome's behavior, which includes per-[script](https://en.wikipedia.org/wiki/Script_(Unicode)) fallbacks.

Fixes #15481.

TODO:
- [x] tests. There's one test for japanese script, but it needs a couple more to make sure the mechanism is working correctly.
- [x] Chromium uses a [cache](https://chromium.googlesource.com/chromium/src/+/69.0.3497.106/chrome/browser/font_family_cache.cc) and claims that iterating the whole list every time adds significantly to renderer startup time. That doesn't intuitively seem true to me, but I haven't measured. It needs to be measured, and possibly we should build a cache.
- [x] ~Evaluate the possibility of directly using `font_family_cache.cc` from Chrome.~ Its use of `Profile` doesn't match the needs of Electron.
- [x] ~Evaluate the possibility of splitting out `kFontDefaults` in upstream Chromium (defined [here](https://chromium.googlesource.com/chromium/src/+/69.0.3497.106/chrome/browser/ui/prefs/prefs_tab_helper.cc#157)) so we can include them without patching.~ This would be difficult currently, these defaults are embedded pretty deep into Chromium's preference system and it'd be a challenge to refactor them to a common place in a way that would make sense to upstream to Chromium. The lists change quite infrequently though: the last change was [in 2016](https://chromium.googlesource.com/chromium/src/+blame/master/chrome/browser/ui/prefs/prefs_tab_helper.cc#125), so I think it's OK to copy/paste them.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [ ] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

#### Release Notes

Notes: fixed default font fallback for non-latin scripts

### CODE REVIEW

**Comment by nornagon** [comment]
I measured the performance of the `SetFontDefaults` function and found it took about 500µs on my mac pro. I think that's slow enough that it's worth adding a cache.

File: atom/browser/font_defaults.cc [hunk 1]
```diff
@@ -0,0 +1,181 @@
+// Copyright (c) 2018 Slack Technologies, Inc.
```
Comment (jkleinsc) [inline atom/browser/font_defaults.cc]: Should there be a Slack copyright here?

Comment (jkleinsc) [reply atom/browser/font_defaults.cc]: I didn't realize that we need copyrights on these files.  It is fine that it has the Slack copyright as it is following the license terms.

File: spec/chromium-spec.js [hunk 2]
```diff
@@ -1312,3 +1313,59 @@ describe('chromium feature', () => {
...
+
+  it('should use Helvetica for sans-serif on Mac, and Arial on Windows and Linux', async () => {
+    const html = `<body style="font-family: sans-serif">test</body>`
+    const fonts = await getRenderedFonts(html)
+    expect(fonts).to.be.an('array')
+    expect(fonts).to.have.length(1)
+    expect(fonts[0].familyName).to.equal({
+      'win32': 'Arial',
+      'darwin': 'Helvetica',
+      'linux': 'DejaVu Sans' // I think this depends on the distro? We don't specify a default.
```
Comment (MarshallOfSound) [inline spec/chromium-spec.js]: Is that really a default font on linux?  I thought you had to install those powerline fonts manually 🤔

Comment (nornagon) [reply spec/chromium-spec.js]: > DejaVu fonts can be obtained from the DejaVu project on GitHub. Some operating systems (OpenBSD, Solaris, Haiku, AmigaOS 4, GNU/Linux distributions such as Ubuntu, Debian, Fedora, and RHEL) include DejaVu fonts in their default installation,[3][4][5][6][7] sometimes even using them as their system fonts

https://en.wikipedia.org/wiki/DejaVu_fonts

**Pull request merged**